functions/Get-DbaDbCompression.ps1

```diff
@@ -0,0 +1,143 @@
+
+        .PARAMETER Database
+            The database(s) to process - this list is auto populated from the server. If unspecified, all databases will be processed.
+
+        .PARAMETER ExcludeDatabase
+            The database(s) to exclude - this list is auto populated from the server.
+
+        .PARAMETER EnableException
+            By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+            This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+            Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+        .NOTES
+            Author: Jess Pomfret (@jpomfret jesspomfret.com)
+            Tags: Compression, Table, Database
+            Website: https://dbatools.io
+            Copyright: (C) Chrissy LeMaire, clemaire@gmail.com
+            License: MIT https://opensource.org/licenses/MIT
+
+        .EXAMPLE
+            Get-DbaDbCompression -SqlInstance localhost
+
+            Returns objects size and current compression level for all user databases.
+
+        .EXAMPLE
+            Get-DbaDbCompression -SqlInstance localhost -Database TestDatabase
+
+            Returns objects size and current compression level for objects within the TestDatabase database.
+
+            .EXAMPLE
+            Get-DbaDbCompression -SqlInstance localhost -ExcludeDatabase TestDatabases
+
+            Returns objects size and current compression level for objects in all databases except the TestDatabase database.
+    #>
+    [CmdletBinding(DefaultParameterSetName = "Default")]
+    param (
+        [parameter(Mandatory = $true, ValueFromPipeline = $true)]
+        [Alias("ServerInstance", "SqlServer")]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [object[]]$Database,
+        [object[]]$ExcludeDatabase,
+        [Alias('Silent')]
+        [switch]$EnableException
+    )
+
+    process {
+        $results = @()
+        foreach ($instance in $SqlInstance) {
+            try {
+                Write-Message -Level VeryVerbose -Message "Connecting to $instance" -Target $instance
+                $server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $SqlCredential -MinimumVersion 10
+            }
+            catch {
+                Stop-Function -Message "Failed to process Instance $Instance" -ErrorRecord $_ -Target $instance -Continue
+            }
+
+            try {
+                $dbs = $server.Databases | Where-Object IsAccessible
+
+                if ($Database) {
+                    $dbs = $dbs | Where-Object { $Database -contains $_.Name -and $_.IsSystemObject -eq 0 }
+                }
+
+                else {
+                    $dbs = $dbs | Where-Object { $_.IsSystemObject -eq 0 }
+                }
+
+                if (Test-Bound "ExcludeDatabase") {
```

Comment:
can just be if ($ExcludeDatabase) and you can/should filter as usual with -in/-notin or -contains/-notcointains ... as long as it's consistent it's nicer to see